packages/connect-thegraph-tokens/subgraph/src/TokenManager.ts

```diff
@@ -23,15 +24,81 @@ export function getTokenManagerEntity(proxyAddress: Address): TokenManagerEntity
     tokenManagerEntity.address = proxyAddress
     tokenManagerEntity.orgAddress = tokenManagerContract.kernel()
 
+    // See if the TokenManager is already initialized with a token.
+    // If it's not, remember it so that we can check later.
     let tokenAddress = tokenManagerContract.token()
-    aragon.processToken(tokenAddress)
+    if (tokenAddress.toHexString() != '0x0000000000000000000000000000000000000000') {
+      aragon.processToken(tokenAddress)
+    } else {
+      _registerOrphanTokenManager(proxyAddress)
+    }
 
     tokenManagerEntity.save()
   }
 
   return tokenManagerEntity!
 }
 
+export function processOrphanTokenManagers(): void {
+  let registry = _getOrphanTokenManagersEntity()
+
+  let apps = registry.apps
+  for (let i = 0; i < apps.length; i++) {
+    let proxyAddress = apps[i] as Address
+
+    let tokenManagerContract = TokenManagerContract.bind(proxyAddress)
+
+    let tokenAddress = tokenManagerContract.token()
+    if (tokenAddress.toHexString() != '0x0000000000000000000000000000000000000000') {
+      _unregisterOrphanTokenManager(proxyAddress)
+
+      aragon.processToken(tokenAddress)
+    }
+  }
+}
+
+function _registerOrphanTokenManager(appAddress: Address): void {
+  let registry = _getOrphanTokenManagersEntity()
+
+  let apps = registry.apps
+  if (!apps.includes(appAddress)) {
+    apps.push(appAddress)
+
+    registry.apps = apps
+
+    registry.save()
+  }
+}
+
+function _unregisterOrphanTokenManager(appAddress: Address): void {
+  let registry = _getOrphanTokenManagersEntity()
+
+  let apps = registry.apps
+  if (apps.includes(appAddress)) {
+    let idx = apps.indexOf(appAddress)
+    apps.splice(idx, 1)
```

Comment:
I wonder if we can do this kind of logic on AssemblyScript?

Reply:
Looks like it works, yeah. This is a trace I did to verify (newest in the bottom):
```
> Unregister tm, Orphans(before): [[object Object]], data_source: DeepDAO, block_hash: 0x5fb5fe5f291530cfffc81267e849fde388462cb436b1d71987f120ac8cdaa12a, block_number: 8134972
> Unregister tm, Orphans(after): [], data_source: DeepDAO, block_hash: 0x5fb5fe5f291530cfffc81267e849fde388462cb436b1d71987f120ac8cdaa12a, block_number: 8134972
> Register tm, Orphans(after): [[object Object]], data_source: DeepDAO, block_hash: 0xbd23eb97c7de5ffc998f19148748ecca4249ddc0bab35916e1b699a5706728d2, block_number: 8134920
> Register tm, Orphans(before): [], data_source: DeepDAO, block_hash: 0xbd23eb97c7de5ffc998f19148748ecca4249ddc0bab35916e1b699a5706728d2, block_number: 8134920
```